.github/workflows/validate-csharp.yml

```
@@ -0,0 +1,77 @@
+name: CSharp
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+env:
+  FORCE_COLOR: 3
+
+jobs:
+  test:
+    name: Test [${{ matrix.os }}]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: 6.0.x
+
+      - name: dotnet test Facebook.Yoga.Tests
+        run: dotnet test
+        working-directory: csharp/tests/Facebook.Yoga.Tests
+
+  build:
+    name: Build and Package [${{ matrix.os }}]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup
+        uses: ./.github/actions/setup-js
```

`setup-js` installs the Emscripten toolchain, which we don't really need. But it seems like we could factor this to instead be a setup script for CS with:
1. "actions/setup-dotnet"
2. "dotnet workload restore" (current CI failures point to needing to add this)

```suggestion
        uses: ./.github/actions/setup-cs
```